functions/source/soci-index-generator-lambda/utils/registry/registry.go

```diff
@@ -143,24 +158,66 @@ func (registry *Registry) GetManifest(ctx context.Context, repositoryName string
 	return manifest, nil
 }
 
-// Validate if a digest is a valid image manifest
-func (registry *Registry) ValidateImageManifest(ctx context.Context, repositoryName string, digest string) error {
-	manifest, err := registry.GetManifest(ctx, repositoryName, digest)
+func (registry *Registry) validateImageManifest(ctx context.Context, repositoryName string, digest string) error {
+	// Get the manifest content
+	imageRef, err := registry.GetDigest(ctx, repositoryName, digest)
```

Comment:
nit:
```suggestion
	manifest, err := registry.GetManifest(ctx, repositoryName, digest)
```
The new names here are not correct and confusing.